File: src/main/java/net/green_jab/fixed_minecraft/FixedMinecraft.java
package net.green_jab.fixed_minecraft;

import net.fabricmc.api.ModInitializer;

import net.green_jab.fixed_minecraft.block.ModBlocks;
import net.green_jab.fixed_minecraft.item.ModItemGroups;
import net.green_jab.fixed_minecraft.item.ModItems;
import net.green_jab.fixed_minecraft.recipe.ModRecipes;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class FixedMinecraft implements ModInitializer {
	public static final String MOD_ID = "fixed_minecraft";//
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

	@Override
	public void onInitialize() {
		LOGGER.info("Hello Fabric world!");
		ModItemGroups.registerItemGroups();

		ModItems.registerItems();
		ModBlocks.registerBlocks();

		ModRecipes.registerRecipes();
	}
}

File: src/main/kotlin/net/greenjab/fixedminecraft/FixedMinecraft.kt
package net.greenjab.fixedminecraft

import net.fabricmc.api.ModInitializer
import net.greenjab.fixedminecraft.network.SyncHandler
import net.greenjab.fixedminecraft.registry.registries.ItemGroupRegistry
import net.greenjab.fixedminecraft.registry.registries.GameruleRegistry
import net.greenjab.fixedminecraft.registry.registries.RecipeRegistry
import net.minecraft.block.DispenserBlock
import net.minecraft.item.Item
import net.minecraft.item.Items
import net.minecraft.server.MinecraftServer
import net.minecraft.util.Identifier
import org.slf4j.LoggerFactory
import java.util.HashMap


object FixedMinecraft : ModInitializer {
    val logger = LoggerFactory.getLogger("FixedMinecraft")
    var SERVER: MinecraftServer? = null

    var netheriteAnvil = false
    var ItemCapacities : HashMap<Item, Int> = hashMapOf()

    //TODO Rename functions / commenting
    //TODO biome colours?
    //TODO Pale garden ominous biome?
    //TODO Remove Kotlin
    //TODO LOTS of bug testing

    override fun onInitialize() {
        logger.info("Initializing ${FixedMinecraftConstants.MOD_NAME}")

        SyncHandler.init()

        ItemGroupRegistry.register()

        RecipeRegistry.register()
        GameruleRegistry.register()

        DispenserBlock.registerProjectileBehavior(Items.BRICK)
        DispenserBlock.registerProjectileBehavior(Items.NETHER_BRICK)
        DispenserBlock.registerProjectileBehavior(Items.RESIN_BRICK)
        DispenserBlock.registerProjectileBehavior(Items.TRIDENT)
    }

    fun id(path: String): Identifier? {
        return Identifier.of(FixedMinecraftConstants.NAMESPACE, path)
    }
}
